src/components/forms/DateRangePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,145 @@
+): React.ReactElement => {
+  const { 
+    startDateLabel,
+    startDateHint,
+    startDatePickerProps,
+    endDateLabel,
+    endDateHint,
+    endDatePickerProps,
+    className
+  } = props
+
+  const [
+    startDateInternalValue,
+    setStartDateInternalValue
+  ] = useState<string | undefined>(startDatePickerProps.defaultValue);
+  const [
+    endDateInternalValue,
+    setEndDateInternalValue
+  ] = useState<string | undefined>(endDatePickerProps.defaultValue);
+  
+  const startDatePickerOnChange = (externallyFormattedValue: string | undefined): void => {
+    const parsedValue = 
+      externallyFormattedValue && parseDateString(externallyFormattedValue, DEFAULT_EXTERNAL_DATE_FORMAT)
+    const internallyFormattedValue = 
+      parsedValue && formatDate(parsedValue)
+    
+    if (parsedValue) setStartDateInternalValue(internallyFormattedValue)
+    if (startDatePickerProps.onChange) startDatePickerProps.onChange(externallyFormattedValue)
+  }
+
+  const endDatePickerOnChange = (externallyFormattedValue: string | undefined): void => {
+    const parsedValue = 
+      externallyFormattedValue && parseDateString(externallyFormattedValue, DEFAULT_EXTERNAL_DATE_FORMAT)
+    const internallyFormattedValue = 
+      parsedValue && formatDate(parsedValue)
+    
+    if (parsedValue) setEndDateInternalValue(internallyFormattedValue)
+    if (endDatePickerProps.onChange) endDatePickerProps.onChange(externallyFormattedValue)
+  }
+
+  const getMaxStartDate = (): string | undefined => {
+    const { maxDate: maxStartDate } = startDatePickerProps
+    const parsedMaxStartDate = maxStartDate && parseDateString(maxStartDate, DEFAULT_EXTERNAL_DATE_FORMAT)
+    const parsedCurrentEndDate = endDateInternalValue && parseDateString(endDateInternalValue)
+
+    if (parsedCurrentEndDate && parsedMaxStartDate) {
+      if (parsedCurrentEndDate.getTime() < parsedMaxStartDate.getTime()) {
+        return formatDate(parsedCurrentEndDate)
+      } else {
+        return formatDate(parsedMaxStartDate)
+      }
+    } else {
+      return (parsedCurrentEndDate && formatDate(parsedCurrentEndDate)) 
+        || (parsedMaxStartDate && formatDate(parsedMaxStartDate)) 
+        || undefined
+    }
+  }
+
+  const getMinEndDate = (): string | undefined => {
+    const { minDate: minEndDate } = endDatePickerProps
+    const parsedMinEndDate = minEndDate && parseDateString(minEndDate, DEFAULT_EXTERNAL_DATE_FORMAT)
+    const parsedCurrentStartDate = startDateInternalValue && parseDateString(startDateInternalValue)
+
+    if (parsedCurrentStartDate && parsedMinEndDate) {
+      if (parsedCurrentStartDate.getTime() > parsedMinEndDate.getTime()) {
+        return formatDate(parsedCurrentStartDate)
+      } else {
+        return formatDate(parsedMinEndDate)
+      }
+    } else {
+      return (parsedCurrentStartDate && formatDate(parsedCurrentStartDate)) 
+        || (parsedMinEndDate && formatDate(parsedMinEndDate)) 
+        || undefined
+    }
+  }
```

Comment:
I've just identified a bug with the logic in these methods. Fixing and adding tests.

Reply:
6180ab6 & 2fedaec fix the issues I had identified.

Reply:
ad82800 adds the test covering these issues so they do not get reintroduced